src/ShaderBlob.cpp

```diff
@@ -193,4 +206,26 @@ bool WritePermutation(
     return success;
 }
 
+
+std::vector<size_t> GetSortedConstantsIndices(
+    const std::string* constants,
```

Comment:
Passing a pointer to an array of std::string looks really weird. Why not a `vector<string>&`?